rust/datafusion/src/physical_plan/parquet.rs

```diff
@@ -209,6 +251,479 @@ impl ParquetExec {
     }
 }
 
+#[derive(Debug, Clone)]
+/// Predicate builder used for generating of predicate functions, used to filter row group metadata
+pub struct PredicateExpressionBuilder {
```

Comment:
nit: probably this doesn't need to be a `pub` struct given that it seems to be tied to the parquet scan implementation

Reply:
I don't feel strongly about this either way, but at the moment it has to be public because it is used as a parameter in `pub ParquetExec::new`

Reply:
I don't feel strongly either way either -- no need to change it

Reply:
thinking some more about this, it could be done by moving the creation of PredicateExpressionBuilder from `ParquetExec::try_from_files`  into `(ExecutionPlan for ParquetExec)::execute`, but then this work would be repeated for each partition, where as currently it's only done once; at this point I don't think it's worth it.